shtdlib.sh

```diff
@@ -1682,8 +1684,36 @@ function strip_space {
     echo -n "${@}" | sed -e 's/^[[:space:]]*//' -e 's/[[:space:]]*$//'
 }
 
-# Usage
-# This function is used to safely edit files for config parameters, etc
+# Load ini file parameter
+# Requires at least two arguments but optionally accepts a third, ini_section
+# If ini_section is specified and multiple sections match an error will be
+# raised. Multiple matches will otherwise be returned
+# To strip leading/trailing whitespace simple pipe to sed -e 's/^[[:space:]]*//g'
+function load_ini_file_parameter {
+    local filename="${1}"
+    local name="${2}"
+    local ini_section="${3:-}"
+    debug 10 "Loading INI file parameter: ${name} from file: ${filename}, optional section ${ini_section}"
+
+    if [ -n "${ini_section}" ]; then
+        #shellcheck disable=SC2086
+        ini_section_match="$(grep -c "\[${ini_section}\]" "${filename}")"
+        if [ "${ini_section_match}" -lt 1 ]; then
+            color_echo red "Unable to find INI section matching ${ini_section}"
+            return 1
+        elif [ "${ini_section_match}" -eq 1 ]; then
+            debug 9 "Found INI section ${ini_section}"
+            sed -n "/\[${ini_section}\]/,/\[/p" "${filename}" | grep -E "^${name}" | awk -F= '{print $2}'
```

Comment:
Why not limit the grep search to just 1 result?

Reply:
good point